From 06cf748485ae5d419e0a96518ae713b7653194f6 Mon Sep 17 00:00:00 2001
From: Edmunt Pienkowsky <roed@onet.eu>
Date: Sun, 30 Aug 2020 17:13:43 +0200
Subject: [PATCH 3/6] build correctly with CONFIG_ENV_IS_NOWHERE macro defined


diff --git a/board/olimex/a10_olinuxino/board.c b/board/olimex/a10_olinuxino/board.c
index ca1aad8a98..c76b8f7216 100644
--- a/board/olimex/a10_olinuxino/board.c
+++ b/board/olimex/a10_olinuxino/board.c
@@ -51,7 +51,14 @@ int show_board_info(void)
 #ifdef CONFIG_BOARD_LATE_INIT
 int board_late_init(void)
 {
-	/*No need to late init*/
+	int pin;
+
+	pin = sunxi_name_to_gpio("PC3");
+
+	gpio_request(pin, "satapwr");
+	gpio_direction_output(pin, 1);
+	/* Give attached sata device time to power-up to avoid link timeouts */
+	mdelay(500);
 
 	return 0;
 }
@@ -69,19 +76,19 @@ int board_fit_config_name_match(const char *name)
 #ifdef CONFIG_ENV_IS_IN_EXT4
 const char *env_ext4_get_dev_part(void)
 {
-		return "0:auto";
+	return "0:auto";
 }
 #endif /* CONFIG_ENV_IS_IN_EXT4 */
 
 #ifdef CONFIG_ENV_IS_IN_FAT
 const char *env_fat_get_dev_part(void)
 {
-			return "0:auto";
-	
+	return "0:auto";
 }
 #endif /* CONFIG_ENV_IS_IN_FAT */
 
-#if CONFIG_ENV_IS_IN_SPI_FLASH
+#ifdef CONFIG_ENV_IS_IN_SPI_FLASH
+#if defined(CONFIG_ENV_IS_IN_EXT4) && defined(CONFIG_ENV_IS_IN_FAT)
 static enum env_location env_get_location_spi(int prio)
 {
 	switch (prio) {
@@ -97,7 +104,6 @@ static enum env_location env_get_location_spi(int prio)
 
 	return ENVL_UNKNOWN;
 }
-#endif
 
 static enum env_location env_get_location_default(int prio)
 {
@@ -112,6 +118,17 @@ static enum env_location env_get_location_default(int prio)
 
 	return ENVL_UNKNOWN;
 }
+#else
+static enum env_location env_get_location_spi(int __maybe_unused prio)
+{
+	return ENVL_SPI_FLASH;
+}
+
+static enum env_location env_get_location_default(int __maybe_unused prio)
+{
+	return ENVL_NOWHERE;
+}
+#endif
 
 enum env_location env_get_location(enum env_operation op, int prio)
 {
@@ -119,17 +136,13 @@ enum env_location env_get_location(enum env_operation op, int prio)
 
 	switch (boot) {
 	case BOOT_DEVICE_BOARD:
-#ifdef CONFIG_ENV_IS_IN_SPI_FLASH
 		if (olinuxino_board_has_spi())
 			return env_get_location_spi(prio);
 		else
-#endif
 			return env_get_location_default(prio);
 
-#ifdef CONFIG_ENV_IS_IN_SPI_FLASH
 	case BOOT_DEVICE_SPI:
 		return env_get_location_spi(prio);
-#endif
 
 	case BOOT_DEVICE_MMC1:
 	case BOOT_DEVICE_MMC2:
@@ -142,3 +155,19 @@ enum env_location env_get_location(enum env_operation op, int prio)
 
 	return env_get_location_default(prio);
 }
+
+#elif defined(CONFIG_ENV_IS_IN_EXT4) && defined(CONFIG_ENV_IS_IN_FAT)
+enum env_location env_get_location(enum env_operation op, int prio)
+{
+	switch (prio) {
+	case 0:
+		return ENVL_EXT4;
+	case 1:
+		return ENVL_FAT;
+	default:
+		break;
+	}
+
+	return ENVL_UNKNOWN;
+}
+#endif
diff --git a/board/olimex/a13_olinuxino/board.c b/board/olimex/a13_olinuxino/board.c
index 0dc878a5f6..14bd37eeaf 100644
--- a/board/olimex/a13_olinuxino/board.c
+++ b/board/olimex/a13_olinuxino/board.c
@@ -89,19 +89,19 @@ int board_fit_config_name_match(const char *name)
 #ifdef CONFIG_ENV_IS_IN_EXT4
 const char *env_ext4_get_dev_part(void)
 {
-		return "0:auto";
+	return "0:auto";
 }
 #endif /* CONFIG_ENV_IS_IN_EXT4 */
 
 #ifdef CONFIG_ENV_IS_IN_FAT
 const char *env_fat_get_dev_part(void)
 {
-			return "0:auto";
-	
+	return "0:auto";
 }
 #endif /* CONFIG_ENV_IS_IN_FAT */
 
-#if CONFIG_ENV_IS_IN_SPI_FLASH
+#ifdef CONFIG_ENV_IS_IN_SPI_FLASH
+#if defined(CONFIG_ENV_IS_IN_EXT4) && defined(CONFIG_ENV_IS_IN_FAT)
 static enum env_location env_get_location_spi(int prio)
 {
 	switch (prio) {
@@ -117,7 +117,6 @@ static enum env_location env_get_location_spi(int prio)
 
 	return ENVL_UNKNOWN;
 }
-#endif
 
 static enum env_location env_get_location_default(int prio)
 {
@@ -132,6 +131,17 @@ static enum env_location env_get_location_default(int prio)
 
 	return ENVL_UNKNOWN;
 }
+#else
+static enum env_location env_get_location_spi(int __maybe_unused prio)
+{
+	return ENVL_SPI_FLASH
+}
+
+static enum env_location env_get_location_default(int __maybe_unused prio)
+{
+	return ENVL_NOWHERE;
+}
+#endif
 
 enum env_location env_get_location(enum env_operation op, int prio)
 {
@@ -139,17 +149,13 @@ enum env_location env_get_location(enum env_operation op, int prio)
 
 	switch (boot) {
 	case BOOT_DEVICE_BOARD:
-#ifdef CONFIG_ENV_IS_IN_SPI_FLASH
 		if (olinuxino_board_has_spi())
 			return env_get_location_spi(prio);
 		else
-#endif
 			return env_get_location_default(prio);
 
-#ifdef CONFIG_ENV_IS_IN_SPI_FLASH
 	case BOOT_DEVICE_SPI:
 		return env_get_location_spi(prio);
-#endif
 
 	case BOOT_DEVICE_MMC1:
 	case BOOT_DEVICE_MMC2:
@@ -162,3 +168,19 @@ enum env_location env_get_location(enum env_operation op, int prio)
 
 	return env_get_location_default(prio);
 }
+
+#elif defined(CONFIG_ENV_IS_IN_EXT4) && defined(CONFIG_ENV_IS_IN_FAT)
+enum env_location env_get_location(enum env_operation op, int prio)
+{
+	switch (prio) {
+	case 0:
+		return ENVL_EXT4;
+	case 1:
+		return ENVL_FAT;
+	default:
+		break;
+	}
+
+	return ENVL_UNKNOWN;
+}
+#endif
diff --git a/board/olimex/a20_olinuxino/board.c b/board/olimex/a20_olinuxino/board.c
index 0130f4e248..394b94be25 100644
--- a/board/olimex/a20_olinuxino/board.c
+++ b/board/olimex/a20_olinuxino/board.c
@@ -190,6 +190,7 @@ const char *env_fat_get_dev_part(void)
 }
 #endif /* CONFIG_ENV_IS_IN_FAT */
 
+#if defined(CONFIG_ENV_IS_IN_EXT4) && defined(CONFIG_ENV_IS_IN_FAT)
 enum env_location env_get_location(enum env_operation op, int prio)
 {
 	switch (prio) {
@@ -203,3 +204,4 @@ enum env_location env_get_location(enum env_operation op, int prio)
 
 	return ENVL_UNKNOWN;
 }
+#endif
-- 
2.28.0

